src/Php/XdebugHandler.php

```diff
@@ -0,0 +1,185 @@
+<?php
+/**
+ * Copyright © 2017 Maks Rafalko
+ *
+ * License: https://opensource.org/licenses/BSD-3-Clause New BSD License
+ */
+declare(strict_types=1);
+
+namespace Infection\Php;
+
+class XdebugHandler
+{
+    const ENV_DISABLE_XDEBUG = 'INFECTION_DISABLE_XDEBUG';
+
+    const RESTART_HANDLE = 'internal';
+
+    /**
+     * @var ConfigBuilder
+     */
+    private $configBuilder;
+
+    /**
+     * @var bool
+     */
+    private $isLoaded;
+
+    /**
+     * @var string
+     */
+    private $envScanDir;
+
+    /**
+     * @var string
+     */
+    private $tmpIniPath;
+
+    public function __construct(ConfigBuilder $configBuilder)
+    {
+        $this->configBuilder = $configBuilder;
+
+        $this->isLoaded = \extension_loaded('xdebug');
+        $this->envScanDir = (string) getenv(ConfigBuilder::ENV_PHP_INI_SCAN_DIR);
+    }
+
+    public function check()
+    {
+        $args = explode('|', (string) getenv(self::ENV_DISABLE_XDEBUG));
+
+        if ($this->needsRestart($args[0])) {
+            if ($this->prepareRestart()) {
```

Comment:
Can the if statements be combined?

Reply:
I left it on purpose. It looks better from readability point of view.